scripts/test_env.py

```diff
@@ -0,0 +1,244 @@
+        self.api_port = 8080 + nth
+        self.telemetry_port = 8180 + nth
+        self.tokio_console_port = 5555 + nth
+        self.out_dir = args.out_dir
+        self.root_dir = args.root_dir
+        self.host_addr = args.host_addr
+
+        logging.info(f"Peer {self.name} generating key pair...")
+
+        kagami = subprocess.run([f"{self.out_dir}/kagami","crypto", "-j"],
+                                capture_output=True)
+        if kagami.returncode:
+            logging.error("Kagami failed to generate a key pair.")
+            sys.exit(3)
+        str_keypair = kagami.stdout
+        json_keypair = json.loads(str_keypair)
+        # public key is a string, private key is a json object
+        self.public_key = json_keypair['public_key']
+        self.private_key = json.dumps(json_keypair['private_key'])
+
+        logging.info(f"Peer {self.name} initialized")
+        
+    def run(self, is_genesis: bool = False):
+        logging.info(f"Running peer {self.name}...")
+        peer_dir = self.out_dir.joinpath(f"peers/{self.name}")
+        os.mkdir(peer_dir)
+        os.mkdir(peer_dir.joinpath("storage"))
+
+        # Copying the binary to have unique names for each peer
+        iroha_bin = f"{peer_dir}/{self.name}"
+        shutil.copy2(f"{self.out_dir}/peers/iroha", iroha_bin)
+
+        os.environ["KURA_BLOCK_STORE_PATH"] = str(peer_dir.joinpath("storage"))
+        os.environ["LOG_FILE_PATH"] = str(peer_dir.joinpath("log.json"))
+        os.environ["MAX_LOG_LEVEL"] = "TRACE"
+        os.environ["IROHA_PUBLIC_KEY"] = self.public_key
+        os.environ["IROHA_PRIVATE_KEY"] = self.private_key
+        os.environ["SUMERAGI_DEBUG_FORCE_SOFT_FORK"] = "false"
+        os.environ["TORII_P2P_ADDR"] = f"{self.host_addr}:{self.p2p_port}"
+        os.environ["TORII_API_URL"] = f"{self.host_addr}:{self.api_port}"
+        os.environ["TORII_TELEMETRY_URL"] = f"{self.host_addr}:{self.telemetry_port}"
+        os.environ["TOKIO_CONSOLE_ADDR"] = f"{self.host_addr}:{self.tokio_console_port}"
+
+        genesis_arg = "--submit-genesis" if is_genesis else ""
+        # FD never gets closed
+        log_file = open(peer_dir.joinpath(".log"), "w")
+        # These processes are created detached from the parent process already
+        subprocess.Popen([iroha_bin, f"{genesis_arg}"],
+                    stdout=log_file, stderr=subprocess.STDOUT)
+
+def pos_int(arg):
+    if int(arg) > 0:
+        return int(arg)
+    else:
+        raise argparse.ArgumentTypeError(f"Argument {arg} must be a positive integer")
+
+def prompt_build_bin(bin_name: str, root_dir: pathlib.Path, target_dir: pathlib.Path):
+    logging.error(f"The binary `{bin_name}` wasn't found in `{root_dir}` directory")
+    while True:
+        prompt = input(f"Build it by running `cargo build --bin {bin_name}`? (Y/n)\n")
+        if prompt.lower() in ["y", "yes", ""]:
+            subprocess.run(
+                ["cargo", "build", "--bin", f"{bin_name}"],
```

Comment:
```suggestion
                ["cargo", "build", "--bin", bin_name],
```

Maybe I miss something...